adit/upload/static/upload/upload.js

```diff
@@ -0,0 +1,361 @@
+
+                if (progBar instanceof HTMLProgressElement) {
+                  progBar.value = (loadedFiles / datasets.length) * 100;
+                }
+              } else {
+                this.uploadResultText = "Upload Failed";
+                this.pbVisible = false;
+                this.uploadCompleteTextVisible = true;
+                this.stopUploadButtonVisible = false;
+                break;
+              }
+            }
+            if (loadedFiles == datasets.length) {
+              this.finishUploadComplete();
+            } else {
+              this.finishUploadIncomplete();
+            }
+          } else {
+            this.uploadResultText = "Upload refused - Fehlerhafte Datensätze";
+            this.buttonVisible = false;
+            this.uploadCompleteTextVisible = true;
+          }
+        } catch (e) {
+          this.uploadResultText = "Upload Failed due to an Error";
+          this.buttonVisible = false;
+          this.uploadCompleteTextVisible = true;
+          console.error(e);
+        }
+      }
+    },
+
+    finishUploadComplete: function () {
+      this.uploadResultText = "Upload Successful!";
+      this.stopUploadButtonVisible = false;
+      this.pbVisible = false;
+      this.uploadCompleteTextVisible = true;
+
+      setTimeout(() => {
+        this.uploadCompleteTextVisible = false;
+      }, 5000);
+    },
+
+    stopUpload: function () {
+      this.stopUploadVar = true;
+      this.stopUploadButtonVisible = false;
+
+      this.finishUploadIncomplete();
+    },
+
+    finishUploadIncomplete: function () {
+      if (this.stopUploadVar) {
+        this.uploadResultText = "Upload Cancelled";
+      } else {
+        this.uploadResultText = "Upload Failed";
+      }
+      this.stopUploadButtonVisible = false;
+      this.pbVisible = false;
+      this.uploadCompleteTextVisible = true;
+
+      setTimeout(() => {
+        this.uploadCompleteTextVisible = false;
+      }, 5000);
+    },
+
+    createAnonymizer: function () {
+      const pseudonym = formEl.querySelector('[name="pseudonym"]');
+      var newPatientID = "";
+      if (pseudonym instanceof HTMLInputElement) {
+        newPatientID = pseudonym.value;
+      }
+      const seed = JSON.parse(
+        document.getElementById("anon-seed-json").textContent
+      );
+      const anon = new Anonymizer(
+        newPatientID,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        seed
+      );
+
+      return anon;
+    },
+
+    retrieveFilefromFileEntry: async function (fileEntry) {
+      return new Promise(fileEntry.file.bind(fileEntry));
+    },
+
+    readDirectory: function (item) {
+      const directoryReader = item.createReader();
+      return new Promise(directoryReader.readEntries.bind(directoryReader));
+    },
+
+    traverseDirectory: async function (item, files) {
+      if (item.isFile) {
+        const file = await this.retrieveFilefromFileEntry(item, files);
```

Comment:
Why is this called with two params when the signature only has one param? Why is it an extra function when it is used no where else?

Reply:
nachschauen ob Promise notwendig ist -> bind nachschauen